File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public bool HasCycle(ListNode head) {
        var slow = head;
        var fast = head?.next;
        while(slow is not null && fast is not null)
        {
            if(slow.val == fast.val)
            {
                return true;
            }
                slow = slow.next;
                fast = fast.next?.next;
        }
        return false;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @param {number} n
     * @return {ListNode}
     */
    removeNthFromEnd(head, n) {
         let fast = head,
    delayedSteps = 0;

  while (fast && fast.next !== null && delayedSteps < n) {
    fast = fast.next;
    delayedSteps++;
  }
  if (delayedSteps === 0) {
    // means that we have one node and will be removed
    // if it pass this cond we have 1 or more
    return null;
  }
  if (delayedSteps === n - 1) {
    return head.next;
  }
  let slow = head;
  while (fast && fast.next !== null) {
    fast = fast.next;
    slow = slow.next;
  }
  if (slow?.next) {
    slow.next = slow.next.next;
  }
  return head;
    }
}
